### DISCUSSION TITLE
Sonoff iHost Smart Home Hub from ITead - Can it be hacked to run Linux operating-system distribution with Zigbee2MQTT or even Home Assistant OS?

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#16983] [General]
Copy/re-posting this from the thread in Home Assistant community forum -> https://community.home-assistant.io/t/sonoff-ihost-smart-home-hub-from-itead-can-it-be-hacked-to-run-home-assistant-os-with-bluetooth-thread-matter-and-zha-integrations/545962

ITead has out of the blue just released "***SONOFF iHost Smart Home Hub***" which if we disregard its intended use simply looks to be a single-board computer under the hood.

* [https://itead.cc/product/sonoff-ihost-smart-home-hub/](https://itead.cc/product/sonoff-ihost-smart-home-hub/)

Can this be hacked to install an alternative operating system like [Home Assistant Operating system](https://github.com/home-assistant/operating-system) or another embedded Linux distro?

Then run Zigbee2MQTT addon or Zigbee2MQTT stand-alone?

https://www.cnx-software.com/2023/03/09/sonoff-ihost-smart-home-hub-enables-local-control-of-sonoff-tasmota-matter-home-automation-devices/

![image](https://user-images.githubusercontent.com/6320001/224299713-62e1ca0b-9280-4d6a-8fff-9e7e62b843b2.png)

I think what should make this all-in-one computer appliance attractive to HA for hacking is that it, similar to the [Home Assistant Yellow](https://yellow.home-assistant.io/), includes a Silicon Labs EFR32MG21 Zigbee/Thread radio module (which is the radio chip that is used in [Home Assistant SkyConnect USB dongle](https://community.home-assistant.io/t/home-assistant-skyconnect-official-radio-usb-dongle-will-be-compatible-with-both-zigbee-and-openthread-including-matter-chip-over-thread/433594/16) and the radio module inside the  [Home Assistant Yellow](https://yellow.home-assistant.io/))., as well as a Realtek RTL8723 WiFi + Bluetooth/BLE module.

So my question to the community is if this could be hacked to run a fork of [Home Assistant Operating System](https://github.com/home-assistant/operating-system) or some other embedded Linux distribution operating system (e.g. stripped [LibreELEC](https://github.com/LibreELEC/LibreELEC.tv) without Kodi/X.Org) which could be used to run either Home Assistant container, Home Assistant core, and Zigbee2MQTT addon?

https://notenoughtech.com/home-automation/sonoff-ihost/

https://www.youtube.com/watch?v=KBvGWoWq__Y&ab_channel=NotEnoughTECH

Internally it uses a carrier-board design (again also similar to [Home Assistant Yellow](https://yellow.home-assistant.io/)) and depending on the model you buy comes with one out of two compute modules which have slightly different specifications for CPU and system RAM-memory:

* SoC and Memory
  * **Rockchip RV1126** ARM Cortex-A7 quad-core CPU @ 1.5 GHz with RISC-V MCU @ 400 MHz, 2D graphics engine, 2.0 TOPS NPU, 4K H.264 and H.265 hardware video decoder and encoder (for accelerated video codec and resolution transcoding), and **4GB DDR4**
      * [https://www.cnx-software.com/2021/01/22/rockchip-rv1126-ai-camera-soc-features-2-0-tops-npu-promises-250ms-fast-boot/](https://www.cnx-software.com/2021/01/22/rockchip-rv1126-ai-camera-soc-features-2-0-tops-npu-promises-250ms-fast-boot/)
  * **Rockchip RV1109** ARM Cortex-A7 dual-core CPU @ 1.5 GHz with RISC-V MCU @ 400 MHz, 2D graphics engine, 1.2 TOPS NPU, 5MP H.264 and H.265 hardware video decoder and encoder (for accelerated video codec and resolution transcoding), and **2GB DDR4**
      * [https://www.cnx-software.com/2021/01/26/rockchip-rv1109-sbc-access-control-solution-4g-lte-connectivity/](https://www.cnx-software.com/2021/01/26/rockchip-rv1109-sbc-access-control-solution-4g-lte-connectivity/)
* **Storage:** 
  * **8GB eMMC flash for the OS**
  * **MicroSD card socket**
* Connectivity
  * 10/100Mbps Ethernet RJ45 port
    * A short RJ-45 Ethernet cable is included in the package
  * Realtek RTL8723 wireless module
    * Bluetooth LE (BLE)
    * WiFi 4 (2.4GHz only)
  * **Silicon Labs EFR32MG21 SoC** for Zigbee 3.0 and Thread/OpenThread
      * [https://www.cnx-software.com/2019/04/26/wireless-gecko-series-2-mcu-extended-range-hw-security/](https://www.cnx-software.com/2019/04/26/wireless-gecko-series-2-mcu-extended-range-hw-security/)
* Audio:
  * Microphone
  * Speaker
* USB – 1x USB 2.0 Type-A host port (which could ex, be used for a Z-Wave Controller USB stick)
* Misc – 8x RGB LEDs
* Power – 5V/2A via USB Type-C port
  * Power-supply is not included (so need to provide own USB-charger that offers 2.1Amp on one port)
* Dimensions – 120 x 41 x 135mm (ABS plastic enclosure)

So to clarify, I am not asking or suggesting to run this as-is with the default operating system firmware that ships from ITead, but instead, if this could be hacked and repurposed as an alternative to the [Home Assistant Yellow](https://yellow.home-assistant.io/) by running a fork of [Home Assistant OS](https://github.com/home-assistant/operating-system) on it?

![image](https://user-images.githubusercontent.com/6320001/224299735-5f5e52e7-03b0-41f8-8039-f9fb3fc58590.png)

![image](https://user-images.githubusercontent.com/6320001/224299751-0f0ecc7d-6a74-4f01-a852-d5e9a4baf91d.png)

![image](https://user-images.githubusercontent.com/6320001/224299771-682a05d3-e084-4dd7-a029-e846e689c065.png)

![image](https://user-images.githubusercontent.com/6320001/224299795-dc2d7cff-57e9-47fe-8c0d-dcd15c39e0d7.png)
PS: For reference, ITead/Sonoff have documentation here for their officially supported OS and firmware:

[https://sonoff.tech/ihost-user-guides/](https://sonoff.tech/ihost-user-guides/)

**Comment by kirovilya** [comment]
we need z2m addon for ihost :)

**Comment by Moatsje** [comment]
Hello hedda, 
have you found a solution to ''hack'' the ihost I have one of my own that I would like to modify.

**Reply by francisp2** [reply]
There is a whole topic about it
https://community.home-assistant.io/t/sonoff-ihost-smart-home-hub-computer-from-itead-can-it-be-hacked-to-run-home-assistant-os-with-bluetooth-thread-matter-and-zha-integrations/545962

**Reply by Hedda** [reply]
Note that they many users are having several issues with the hardware (even when using the stock operating system on it):

https://community.home-assistant.io/t/sonoff-ihost-smart-home-hub-computer-from-itead-can-it-be-hacked-to-run-home-assistant-os-with-bluetooth-thread-matter-and-zha-integrations/545962

So for anyone who has not bought one I would instead recommend buying either a Raspberry Pi or the Home Assistant Green (and use those with an external Zigbee Coordinator USB adapter)

**Reply by humingchun** [reply]
@kirovilya @Moatsje @francisp2 @Hedda Yes, there are ways to run Linux distribution or even Home Assistant on SONOFF iHost. There is an [iHost Open Source Project](https://github.com/iHost-Open-Source-Project) dedicated to this goal.

- To run Linux OS (Prebuilt Linux distro or your own Linux distro), please refer to https://github.com/iHost-Open-Source-Project/operating-system.
- To run Home Assistant (without hacking), please refer to https://github.com/iHost-Open-Source-Project/ha-operating-system.

**Comment by Moatsje** [comment]
Thnx for your reaction!


________________________________
Van: Hedda ***@***.***>
Verzonden: maandag 18 maart 2024 15:07
Aan: Koenkk/zigbee2mqtt ***@***.***>
CC: Moatsje ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [Koenkk/zigbee2mqtt] Sonoff iHost Smart Home Hub from ITead - Can it be hacked to run Linux operating-system distribution with Zigbee2MQTT or even Home Assistant OS? (Discussion #16983)


Note that they many users are having several issues with the hardware:

https://community.home-assistant.io/t/sonoff-ihost-smart-home-hub-computer-from-itead-can-it-be-hacked-to-run-home-assistant-os-with-bluetooth-thread-matter-and-zha-integrations/545962

So for anyone who has not bought one I would instead recommend buying either a Raspberry Pi or the Home Assistant Green (and use those with an external Zigbee Coordinator USB adapter)
